### ISSUE TITLE
Is GenerateInteger Max value inclusive or exclusive?

### ISSUE THREAD

**Issue by ChristopheDEBOVE** [42skillz/Diverse#8]
Is it normal that documentation tells exclusive for maximum in the summary

"Generates a random integer value between a min (inclusive) and a max (exclusive) value."

But inclusive for param
"The inclusive upper bound of the random number returned."

https://github.com/tpierrain/Diverse/blob/0f0b99a1300b4145651cc40bf4c2c03e6789121a/Diverse/Numbers/NumberFuzzer.cs#L25

**Comment by GTechene** [comment]
I agree, the documentation is misleading (there is an explicit test in the repo that actually checks for the inclusivity of the maxValue.

**Comment by aep-sunlife** [comment]
The ambiguity will be a deep source of bugs, in a testing framework no less, until it gets resolved in the documentation.

It's not even clear in this issue discussion what the behavior is.

Best guess, the source code seems to match (Java derived) Random next int behavior, which is closed (inclusive) on the lower bound and open (exclusive) on the upper bound.
